src/wagtail_ai/forms.py

```diff
@@ -35,17 +39,41 @@ def errors_for_json_response(self) -> str:
 
 
 class PromptForm(ApiForm):
-    text = PromptTextField()
+    context = forms.JSONField(required=False, decoder=PromptJSONDecoder)
+    text = PromptTextField(required=False)
     prompt = PromptUUIDField()
 
+    def clean(self):
+        cleaned_data = super().clean()
+        context: PromptContext | None = cleaned_data.get("context")
+        text: str | None = cleaned_data.get("text")
+        prompt: Prompt | None = cleaned_data.get("prompt")
+        # If no context is provided, the form is likely used for features that
+        # assume the input text is to be processed, so text is required.
+        if context is None and not text:
+            self.add_error(
+                "text",
+                ValidationError(
+                    self.fields["text"].error_messages["required"], code="required"
+                ),
+            )
+        elif context is not None and prompt:
+            try:
+                context.clean(prompt.prompt_value)
+            except ValidationError as e:
+                self.add_error("context", e)
+
+        return cleaned_data
```

Comment:
Eventually I think we should drop `text` from the form, so it's just going to be the `prompt` UUID and `context` JSONField. If the prompt needs the input text, it should include something like `{input}` somewhere in the prompt string.